src/osha/oira/content/browser/survey_publication_count.py

```diff
@@ -0,0 +1,60 @@
+from io import StringIO
+from plone import api
+from Products.Five import BrowserView
+
+import csv
+
+
+fieldnames = ["country", "path", "count"]
+
+
+def get_survey_details():
+    details = []
+    pc = api.portal.get_tool("portal_catalog")
+    wft = api.portal.get_tool("portal_workflow")
+    surveys = pc.searchResults(portal_type="euphorie.survey")
+    for brain in surveys:
+        survey = brain.getObject()
+        review_history = wft.getInfoFor(survey, "review_history")
+        count = 0
+        for entry in review_history:
+            if entry["action"] == "publish":
+                count += 1
+        if count:
+            split_path = brain.getPath().split("/")
+            country = split_path[3]
+            details.append(
+                {
+                    "country": country,
+                    "path": "/".join(split_path[2:]),
+                    "count": count,
+                }
+            )
+    sorted_details = sorted(details, key=lambda r: r["country"])
+    return sorted_details
+
+
+class SurveyPublicationCount(BrowserView):
+    """Report the number of times each Survey has been published"""
+
+    def __call__(self):
+        self.fieldnames = fieldnames
+        self.rows = get_survey_details()
+        return self.index()
+
+
+class SurveyPublicationCountCSV(BrowserView):
```

Comment:
This should become:
```suggestion
class SurveyPublicationCountCSV(SurveyPublicationCount):
```
or a method (e.g. `csv` of `SurveyPublicationCount`)